app/grandchallenge/components/signals.py

```diff
@@ -0,0 +1,13 @@
+from django.db.models.signals import pre_delete
+from django.dispatch import receiver
+
+from grandchallenge.algorithms.models import AlgorithmImage, AlgorithmModel
+from grandchallenge.evaluation.models import EvaluationGroundTruth, Method
+
+
+@receiver(pre_delete, sender=AlgorithmImage)
+@receiver(pre_delete, sender=AlgorithmModel)
+@receiver(pre_delete, sender=EvaluationGroundTruth)
+@receiver(pre_delete, sender=Method)

```

Comment:
`WorkstationImage` is missing. Shouldn't the sender be the two abstract classes instead so that we do not have to keep this up to date?

Reply:
The sender can't be an abstract model, as far as I know. Since abstract models do not correspond to actual database tables, they cannot directly be the sender for signals tied to database actions like pre_delete. I will add the WorkstationImage here though

Reply:
True, that works! Thanks. However, I realized that deleting the ComponentImage requires a change to the save method, where we currently check if the image has changed, and if so throw RunTimeError. I'm not sure if we want to change this because it would then open up the possibility that the image gets deleted when the instance does not. So maybe we should go through the client after all?

Reply:
What about setting `save=False` and/or using the `post_delete` signal? https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.fields.files.FieldFile.delete

Reply:
Ah nice, that does the trick